biomage/account/account.py

```diff
@@ -34,99 +34,124 @@ def generate_password():
     )
 
 
-def create_account(full_name, email, userpool):
+def create_account(full_name, email, aws_profile, region, userpool):
     """
     Creates a new account with the information provided.
     Requires a password change call afterwards."""
-    p = Popen(
-        f"""aws cognito-idp admin-create-user \
-            --user-pool-id "{userpool}" \
-            --username "{email}" \
-            --message-action "SUPPRESS" \
-            --user-attributes \
-                "Name=email,Value={email}" \
-                "Name=name,Value='{full_name}'" \
-                "Name=email_verified,Value=true" """,
-        stdin=PIPE,
-        stdout=PIPE,
-        stderr=PIPE,
-        shell=True,
-    )
-
-    output, error = p.communicate(input=b"\n")
-    if output:
-        print(output)
 
-    return error
+    session = boto3.Session(profile_name=aws_profile, region_name=region)
+    cognito = session.client("cognito-idp")
+
+    cognito.admin_create_user(
+        UserPoolId=userpool,
+        Username=email,
+        MessageAction="SUPPRESS",
+        UserAttributes=[
+            {"Name": "email", "Value": email},
+            {"Name": "name", "Value": full_name},
+            {"Name": "email_verified", "Value": "true"},
+        ],
+    )
 
 
 @click.command()
 @click.option(
+    "-e",
     "--email",
     required=True,
     help="User email for the account to change the password in production.",
 )
 @click.option(
+    "-w",
     "--password",
     required=True,
-    help="Password for the new account.",
+    help="New password for the account.",
 )
 @click.option(
+    "-p",
+    "--aws_profile",
+    required=False,
+    default=DEFAULT_AWS_PROFILE,
+    show_default=True,
+    help="The name of the profile stored in ~/.aws/credentials to use.",
+)
+@click.option(
+    "-r",
+    "--region",
+    required=False,
+    default="eu-west-1",
+    show_default=True,
+    help="Region the userpool is in.",
+)
+@click.option(
+    "-u",
     "--userpool",
     required=True,
     help="Userpool of the account to change.",
 )
-def change_password(email, password, userpool):
+def change_password(email, password, aws_profile, region, userpool):
```

Comment:
Is there anty chance you could take the region from the aws_profile?

Reply:
region removed, using default region in `~/.aws/config`